From 9cd50c60f40eabc5ce1583c90f63ddcafb106fe9 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Tue, 13 Aug 2019 14:15:02 +0200
Subject: [PATCH] arm64: dt: fsl-imx8mq-tqma8mq-mba8mx: prevent a lot of dummy
 regulator

add logical supply for following chips from mba8mx 3v3:

- enet phy
- gpio expander
- pwm beeper
- epdev_on for pcie

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx.dts
index 34739572f6cd..b47f46ad12a4 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq-mba8mx.dts
@@ -27,6 +27,7 @@
 		compatible = "pwm-beeper";
 		pwms = <&pwm4 0 250000 0>;
 		beeper-hz = <4000>;
+		amp-supply = <&reg_vcc_3v3>;
 		status = "okay";
 	};
 
@@ -95,6 +96,7 @@
 	pinctrl-0 = <&pinctrl_fec1>;
 	phy-mode = "rgmii-id";
 	phy-handle = <&ethphy0>;
+	phy-supply = <&reg_vcc_3v3>;
 	fsl,magic-packet;
 	phy-reset-gpios = <&expander2 7 GPIO_ACTIVE_LOW>;
 	phy-reset-duration = <500>;
@@ -251,6 +253,7 @@
 		reg = <0x24>;
 		gpio-controller;
 		#gpio-cells = <2>;
+		vcc-supply = <&reg_vcc_3v3>;
 	};
 
 	expander2: pca9555@25 {
@@ -258,6 +261,7 @@
 		reg = <0x25>;
 		gpio-controller;
 		#gpio-cells = <2>;
+		vcc-supply = <&reg_vcc_3v3>;
 	};
 };
 
@@ -315,6 +319,7 @@
 &pcie0{
 	ext_osc = <1>;
 	hard-wired = <1>;
+	epdev_on-supply = <&reg_vcc_3v3>;
 	status = "okay";
 };
 
@@ -322,6 +327,7 @@
 	disable-gpio = <&expander2 13 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&expander2 14 GPIO_ACTIVE_LOW>;
 	ext_osc = <1>;
+	epdev_on-supply = <&reg_vcc_3v3>;
 	status = "okay";
 };
 
